### PR TITLE
#356: [Disk Manager] add large tests for proxy overlay disks

### PR DESCRIPTION
**Pull request by BarkovBG**
https://github.com/ydb-platform/nbs/issues/356

### CODE REVIEW

File: cloud/disk_manager/internal/pkg/facade/disk_service_test/disk_service_test.go [hunk 1]
```diff
@@ -1201,3 +1202,133 @@ func TestDiskServiceCreateEncryptedDiskFromImage(t *testing.T) {
 
 	testcommon.CheckConsistency(t, ctx)
 }
+
+func TestDiskServiceReadFromProxyOverlayDisk(t *testing.T) {
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/facade/disk_service_test/disk_service_test.go:1206]: These tests do not belong here.
Move them to internal/pkg/clients/nbs please

File: cloud/disk_manager/internal/pkg/clients/nbs/testcommon/common.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+func FillDisk(
+	nbsClient nbs.Client,
+	diskID string,
+	diskSize uint64,
+) (uint32, uint64, error) {
+
+	return FillEncryptedDisk(nbsClient, diskID, diskSize, nil)
+}
+
+func FillEncryptedDisk(
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/testcommon/common.go:36]: Add ctx to the signature and move whole method to clients/nbs/testing_client.go, please

File: cloud/disk_manager/internal/pkg/clients/nbs/testcommon/common.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		_, err = acc.Write(data)
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+
+	return acc.Sum32(), storageSize, nil
+}
+
+func GoWriteRandomBlocksToNbsDisk(
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/testcommon/common.go:97]: Also move to testing_client.go

File: cloud/disk_manager/internal/pkg/clients/nbs/tests/client_test.go [hunk 4]
```diff
@@ -1429,3 +1429,114 @@ func TestGetChangedBlocksForLightCheckpoints(t *testing.T) {
...
+		"cp",
+	)
+	require.True(t, created)
+	require.NoError(t, err)
+
+	err = client.ValidateCrc32(
+		ctx,
+		proxyOverlayDiskID,
+		diskContentInfo,
+	)
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/tests/client_test.go:1477]: Fits within 80 symbols

File: cloud/disk_manager/internal/pkg/clients/nbs/tests/client_test.go [hunk 5]
```diff
@@ -1429,3 +1429,114 @@ func TestGetChangedBlocksForLightCheckpoints(t *testing.T) {
...
+		Kind:            types.DiskKind_DISK_KIND_SSD,
+		PartitionsCount: 2,
+	})
+	require.NoError(t, err)
+
+	diskContentInfo, err := client.FillDisk(
+		ctx,
+		diskID,
+		uint64(diskSize),
+	)
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/tests/client_test.go:1501]: Also fits within 80 symbols

File: cloud/disk_manager/internal/pkg/clients/nbs/tests/client_test.go [hunk 6]
```diff
@@ -1429,3 +1429,114 @@ func TestGetChangedBlocksForLightCheckpoints(t *testing.T) {
...
+		PartitionsCount:      1,
+		IsSystem:             true,
+	})
+	require.NoError(t, err)
+
+	err = client.ValidateCrc32(
+		ctx,
+		proxyOverlayDiskID,
+		diskContentInfo,
+	)
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/clients/nbs/tests/client_test.go:1540]: Fits within 80 symbols

**Review by SvartMetal — APPROVED** [review]

**Review by leftmain — APPROVED** [review]

**Pull request merged**